hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/PrefixManagerImpl.java

```diff
@@ -235,33 +235,37 @@ public OMPrefixAclOpResult addAcl(OzoneObj ozoneObj, OzoneAcl ozoneAcl,
     // No explicit prefix create API, both add/set Acl can get new prefix
     // created. When new prefix is created, it should inherit parent prefix
     // or bucket default ACLs.
-    boolean newPrefix = false;
-    if (prefixInfo == null) {
-      OmPrefixInfo.Builder prefixInfoBuilder =
-          new OmPrefixInfo.Builder()
-          .setName(ozoneObj.getPath());
-      if (transactionLogIndex > 0) {
-        prefixInfoBuilder.setObjectID(OmUtils.getObjectIdFromTxId(
-            metadataManager.getOmEpoch(), transactionLogIndex));
-        prefixInfoBuilder.setUpdateID(transactionLogIndex);
-      }
-      prefixInfo = prefixInfoBuilder.build();
-      newPrefix = true;
+    boolean newPrefix = prefixInfo == null;
+    OmPrefixInfo.Builder prefixInfoBuilder = newPrefix
+        ? new OmPrefixInfo.Builder().setName(ozoneObj.getPath())
+        : prefixInfo.toBuilder();
+
+    if (newPrefix && transactionLogIndex > 0) {
+      prefixInfoBuilder.setObjectID(OmUtils.getObjectIdFromTxId(
+          metadataManager.getOmEpoch(), transactionLogIndex));
+      prefixInfoBuilder.setUpdateID(transactionLogIndex);
     }
 
-    boolean changed = prefixInfo.addAcl(ozoneAcl);
+    List<OzoneAcl> updatedAcls = new ArrayList<>(
+        prefixInfo == null ? EMPTY_ACL_LIST : prefixInfo.getAcls());
+    boolean changed = OzoneAclUtil.addAcl(updatedAcls, ozoneAcl);
+
     // Update the in-memory prefix tree regardless whether the ACL is changed.
     // Under OM HA, update ID of the prefix info is updated for every request.
     if (newPrefix) {
-      inheritParentAcl(ozoneObj, prefixInfo);
+      inheritParentAcl(ozoneObj, updatedAcls);
     }
+    OmPrefixInfo updatedPrefixInfo = prefixInfoBuilder
+        .setAcls(updatedAcls)
+        .build();
     // update the in-memory prefix tree
-    prefixTree.insert(ozoneObj.getPath(), prefixInfo);
+    prefixTree.insert(ozoneObj.getPath(), updatedPrefixInfo);
 
     if (!isRatisEnabled) {
-      metadataManager.getPrefixTable().put(ozoneObj.getPath(), prefixInfo);
+      metadataManager.getPrefixTable().put(ozoneObj.getPath(),
+          updatedPrefixInfo);
```

Comment:
nit: no need to wrap (also in `removeAcl` and `setAcl`)

```suggestion
      metadataManager.getPrefixTable().put(ozoneObj.getPath(), updatedPrefixInfo);
```